server/src/main/java/org/elasticsearch/action/admin/indices/rollover/RolloverResponse.java

```diff
@@ -83,7 +102,7 @@ public String getNewIndex() {
     /**
      * Returns the statuses of all the request conditions
      */
-    public Set<Map.Entry<String, Boolean>> getConditionStatus() {
+    public Map<String, Boolean> getConditionStatus() {
```

Comment:
+1 to remove the set, just wondering if this is a a client facing java API change, in which case we should leave the old method on backporting to 6.x

Reply:
yes it is a public facing API yet users have a compiler and can easily figure this out. This set of map entries is probably something that leaves users wonder "why not a map?" anyways.